conda/notices/cache.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+Handles all caching logic including:
+  - Retrieving from cache
+  - Saving to cache
+  - Determining whether not certain items have expired and need to be refreshed
+"""
+import json
+import logging
+import os
+from datetime import datetime, timezone
+from functools import wraps
+from pathlib import Path
+from typing import Optional, Sequence, Set
+
+from conda._vendor.appdirs import user_cache_dir
+from conda.base.constants import APP_NAME, NOTICES_CACHE_SUBDIR, NOTICES_CACHE_FN
+from conda.utils import ensure_dir_exists, safe_open
+
+from .types import ChannelNoticeResponse, ChannelNotice
+
+logger = logging.getLogger(__name__)
+
+
+def cached_response(func):
```

Comment:
I like this idea. @dholth  and I were working together on PR that would see us adding this as a requirement to `conda` (https://github.com/conda/conda/pull/11447), so we have at least two features that would benefit from this.

@jezdez , should we go ahead and add this as a dependency and do the refactor? I'd be more than happy to do that.

Reply:
It's a pretty nice package, I wish the "interpret cache headers" part was a little more separate from the "do the caching" part but the whole library works well. I overrode the cache to delete the contents from cached responses so that I can store them in files, when caching files that I also want to download instead of only retrieve from cache.

Reply:
This is fine, let's merge this as-is and then refactor the caching bits when we get to do #11447 using requests-cache. Thanks for the feedback both!